torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -580,6 +580,65 @@ class AsyncAllreduceWork : public ProcessGroupGloo::AsyncWork {
   }
 };
 
+class AsyncAllreduceCoalescedWork: public ProcessGroupGloo::AsyncWork {
```

Comment:
feel like inherit AsyncAllreduceWork will be more cleaner

Reply:
yeah that's actually the way I implemented it at first, but wasn't sure what the preference would be. I think this way makes more sense after thinking about it some more. Changed.